pyxem/signals/beam_shift.py

```diff
@@ -212,6 +214,61 @@ def get_bivariate_histogram(
         )
         return s_hist
 
+    @property
+    def beam_energy(self):
+        try:
+            return self.metadata.get_item("Acquisition_instrument.TEM.beam_energy")
+        except KeyError:
+            return None
+
+    @beam_energy.setter
+    def beam_energy(self, beam_energy):
+        self.metadata.set_item("Acquisition_instrument.TEM.beam_energy", beam_energy)
+        self.metadata.set_item(
+            "Acquisition_instrument.TEM.wavelength",
+            get_electron_wavelength(beam_energy),
+        )
+
+    def calibrate_electric_shifts(self, thickness):
+        """
+        Calibrate the shifts into the electric field
+
+        m_0 * gamma * |v|^2 * theta / e*t = E
+
+        where m_0 is the electron rest mass, gamma is the Relativistic factor, t is the
+        thickness m_0 is the electron rest mass, e is the electron charge, and theta
+        is the deflection angle in radians.
+
+        Parameters
+        ----------
+        thickness : float
+            The thickness of the sample in nm.
+        """
+        if self.units != ["mrad", "mrad"]:
+            raise ValueError(
+                "The beam shifts must be in mrad units to calibrate to electric field."
+                "Please call `pixels_to_calibrated_units` first."
+            )
+        if self.beam_energy is None:
+            raise ValueError(
+                "The beam energy must be set in the metadata to calibrate to electric field."
+                "Please set the beam energy using `beam_energy` property."
+            )
+        thickness = thickness * 1e-9  # Convert thickness from nm to m
+        m_0 = 9.10938356e-31  # kg
+        e = 1.602176634e-19  # C
+        c = 2.99792458e8  # m/s, speed of light
```

Comment:
While I don't doubt these are correct, you can also find them in scipy.constants

Reply:
@viljarjf good point :)